### PR TITLE
Add mips layers

### PR DESCRIPTION
**Pull request by nullptr-leo**
Add more mips layers.

### CODE REVIEW

**Comment by nihui** [comment]
```
        ncnn::FloatInt fi_bias = { .f = bias };
        v4f32 _bias = (v4f32)__msa_fill_w(fi_bias.i);
```
That seems quite awkward.

how about defining a macro or function __msa_fill_w_f32 with float value as the parameter

**Comment by nullptr-leo** [comment]
> ```
>         ncnn::FloatInt fi_bias = { .f = bias };
>         v4f32 _bias = (v4f32)__msa_fill_w(fi_bias.i);
> ```
> 
> That seems quite awkward.
> 
> how about defining a macro or function __msa_fill_w_f32 with float value as the parameter
> 
> ```
>         v4f32 _bias = (v4f32)__msa_fill_w_f32(bias);
> ```

直接reinterpret_cast吧

**Comment by nihui** [comment]
> > ```
> >         ncnn::FloatInt fi_bias = { .f = bias };
> >         v4f32 _bias = (v4f32)__msa_fill_w(fi_bias.i);
> > ```
> > 
> > 
> > That seems quite awkward.
> > how about defining a macro or function __msa_fill_w_f32 with float value as the parameter
> > ```
> >         v4f32 _bias = (v4f32)__msa_fill_w_f32(bias);
> > ```
> 
> 直接reinterpret_cast吧

好！

**Comment by nihui** [comment]
ping

Is thers any updates ?

**Comment by nullptr-leo** [comment]
> ping
> 
> Is thers any updates ?

暂时没板子，写好的我先整理整理吧

**Comment by nullptr-leo** [comment]
> ping
> 
> Is thers any updates ?

fixed. 其他部分测不了，暂时不提交了

File: src/layer/mips/clip_mips.cpp [hunk 1]
```diff
@@ -42,8 +43,11 @@ int Clip_mips::forward_inplace(Mat& bottom_top_blob, const Option& opt) const
 #endif // __MIPS_MSA
 
 #if __MIPS_MSA
-        v4f32 _max = (v4f32)__msa_fill_w(max);
-        v4f32 _min = (v4f32)__msa_fill_w(min);
+        ncnn::FloatInt fi_max = { .f = max };
+        ncnn::FloatInt fi_min = { .f = min };
+
+        v4f32 _max = (v4f32)__msa_fill_w(fi_max.i);
+        v4f32 _min = (v4f32)__msa_fill_w(fi_min.i);
```
Comment (nihui) [inline src/layer/mips/clip_mips.cpp]: __msa_fill_w_f32 here

**Comment by nihui** [comment]
Remove the WIP tag in title when you think the PR is ready

**Pull request merged**